### PR TITLE
Update auction/swap design to match Sam's mockups

### LINKED ISSUE #1277: Update UI to match latest designs from Sam
https://www.figma.com/design/O1wP7EnPVlTx9GMUFfQA7a/Mini-Front----Swap%2FAuction?node-id=215-2322&t=bRylLPUYsislmLqs-1

### PR DESCRIPTION
**Pull request by jessepinho**
This PR addresses a number of [design tweaks Sam made](https://www.figma.com/design/O1wP7EnPVlTx9GMUFfQA7a/Mini-Front----Swap%2FAuction?node-id=215-2322&t=bRylLPUYsislmLqs-1) to the swap/auction UI:

- Redesign the swap routes
- Redesign the filter UI for auctions
- Add "segments" to the swap duration slider
- Show a tooltip on the duration slider as you drag it
- Redesign the UI for showing filled/unfilled/price impact for instant swaps

https://github.com/penumbra-zone/web/assets/1121544/44938526-6487-455b-b66b-01ab670323fc

Closes #1277

### CODE REVIEW

File: apps/minifront/src/components/swap/auction-list/filters.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    label: 'Upcoming',
+    value: 'upcoming',
+  },
+  {
+    label: 'All',
+    value: 'all',
+  },
+];
+
+export const Filters = () => {
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/auction-list/filters.tsx:27]: image

File: apps/minifront/src/components/swap/duration-slider.tsx [hunk 2]
```diff
@@ -42,16 +45,15 @@ export const DurationSlider = () => {
       </div>
 
       {duration === 'instant' && (
-        <div className='flex flex-col items-center'>
-          Now
-          <span className='text-xs text-muted-foreground'>single trade at market price</span>
+        <div>
+          Now &bull; <span className='text-muted-foreground'>single trade at market price</span>
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/duration-slider.tsx:49]: image

File: apps/minifront/src/components/swap/duration-slider.tsx [hunk 3]
```diff
@@ -42,16 +45,15 @@ export const DurationSlider = () => {
...
         </div>
       )}
 
       {duration !== 'instant' && (
-        <div className='flex flex-col items-center'>
-          ~ {duration}{' '}
-          <span className='text-xs text-muted-foreground'>
+        <div>
+          ~ {duration} &bull;{' '}
+          <span className='text-muted-foreground'>
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/duration-slider.tsx:56]: image

File: apps/minifront/src/components/swap/swap-form/output/estimated-output-explanation.tsx [hunk 4]
```diff
@@ -24,8 +24,9 @@ export const EstimatedOutputExplanation = () => {
   const asssetInSymbol = getSymbolFromValueView.optional()(assetIn?.balanceView);
 
   return (
-    <div className='text-xs text-muted-foreground'>
-      Based on the current estimated market price of {formattedAmount} {assetOut?.symbol} for{' '}
+    <div className='text-xs'>
+      <span className='text-muted-foreground'>Based on the current estimated market price of </span>
+      {formattedAmount} {assetOut?.symbol} <span className='text-muted-foreground'>for </span>
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-form/output/estimated-output-explanation.tsx:29]: image

File: apps/minifront/src/components/swap/swap-form/simulate-swap-result/index.tsx [hunk 5]
```diff
@@ -1,49 +1,57 @@
...
+        Filling <ValueViewComponent view={output} size='sm' /> causes a price impact of{' '}
+        <PriceImpact amount={priceImpact} />.
         {hasUnfilled && (
-          <div className='flex flex-col items-center'>
-            <ValueViewComponent view={unfilled} size='sm' />
-            <span className='font-mono text-[12px] italic text-gray-500'>Unfilled amount</span>
-          </div>
+          <>
+            <ValueViewComponent view={unfilled} size='sm' /> will remain unfilled.
+          </>
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-form/simulate-swap-result/index.tsx:50]: image

File: apps/minifront/src/components/swap/swap-form/simulate-swap-result/price-impact.tsx [hunk 6]
```diff
@@ -7,8 +8,8 @@ export const PriceImpact = ({ amount = 0 }: { amount?: number }) => {
   const percent = formatNumber(amount * 100, { precision: 3 });
 
   return (
-    <div className={cn('flex flex-col text-gray-500 text-sm', amount < -0.1 && 'text-orange-400')}>
-      {percent}%
-    </div>
+    <Pill>
+      <span className={cn('leading-[15px]', amount < -0.1 && 'text-orange-400')}>{percent}%</span>
+    </Pill>
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-form/simulate-swap-result/price-impact.tsx:13]: image

image

File: apps/minifront/src/components/swap/swap-form/simulate-swap-result/traces/index.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import { SwapExecution_Trace } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/dex/v1/dex_pb';
+import { Trace } from './trace';
+import {
+  Metadata,
+  ValueView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { ValueViewComponent } from '@penumbra-zone/ui/components/ui/tx/view/value';
+import { ArrowDown, ArrowUp } from 'lucide-react';
+
+export const Traces = ({
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-form/simulate-swap-result/traces/index.tsx:10]: image

File: packages/ui/components/ui/popover-menu.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  tags: ['autodocs'],
+  argTypes: {
+    trigger: { control: false },
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof PopoverMenu>;
+
+export const Basic: Story = {
```
Comment (jessepinho) [inline packages/ui/components/ui/popover-menu.stories.tsx:19]: image

File: packages/ui/components/ui/popover-menu.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+ * ```tsx
+ * <PopoverMenu
+ *   items={filterOptions}
+ *   value={filter}
+ *   onChange={setFilter}
+ *   trigger={<ListFilter size={16} />}
+ * />
+ * ```
+ */
+export const PopoverMenu = <ValueType extends { toString: () => string }>({
```
Comment (jessepinho) [inline packages/ui/components/ui/popover-menu.tsx:23]: image

File: packages/ui/components/ui/slider.tsx
Comment (jessepinho) [inline packages/ui/components/ui/slider.tsx:1]: moved to `ui/slider/index.tsx`

File: packages/ui/components/ui/slider/index.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  tags: ['autodocs'],
+  argTypes: {
+    thumbTooltip: { control: false },
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof Slider>;
+
+export const Basic: Story = {
```
Comment (jessepinho) [inline packages/ui/components/ui/slider/index.stories.tsx:18]: image

File: packages/ui/components/ui/slider/index.tsx [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+ *   min={0}
+ *   max={10}
+ *   step={2}
+ *   value={value}
+ *   onValueChange={onValueChange}
+ *   segmented // to show gaps in the slider for each option
+ * />
+ * ```
+ */
+const Slider = ({
```
Comment (jessepinho) [inline packages/ui/components/ui/slider/index.tsx:58]: Git is treating this as a new file, but it's actually moved from `../slider.tsx`, and given some new features:

- The `segmented` prop (see screenshot below)
- The `thumbTooltip` prop for rendering a tooltip over the "thumb" of the slider.

Here's the `<DurationSlider />` when using the `segmented` prop:
image


And without the `segmented` prop:

image

File: packages/ui/components/ui/slider/index.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+  segmented,
+  value,
+  onValueChange,
+  min,
+  max,
+  step,
+  thumbTooltip,
+}: PropsWithSegmented | PropsWithoutSegmented) => {
+  const segmentCount =
+    !!segmented && (max - min + 1) % step === 0 ? Math.max((max - min + 1) / step - 1, 1) : 1;
```
Comment (grod220) [inline packages/ui/components/ui/slider/index.tsx:69]: For future maintainability, can we somehow extract this out and/or annotate what this is doing?

**Review by grod220 — APPROVED** [review]
Nice!

File: apps/minifront/src/components/swap/auction-list/filters.tsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  {
+    label: 'Upcoming',
+    value: 'upcoming',
+  },
+  {
+    label: 'All',
+    value: 'all',
+  },
+];
```
[the remark this answers is not in the document]
Comment (jessepinho) [reply apps/minifront/src/components/swap/auction-list/filters.tsx:25]: I mean, we _could_ 😆 I feel like, with just 3 hard-coded options, it's more readable to write it out explicitly. And `PopoverMenuItem` is designed to be flexible, using arbitrary types for the values that may or may not have any relationship to the label, so I personally prefer to keep it this way.

File: packages/ui/components/ui/slider/index.tsx (hunk 12 above)
Comment (jessepinho) [reply packages/ui/components/ui/slider/index.tsx]: Ah, yeah, big oversight on my part 😆 I extracted this logic to a verbose `getSegmentCount` helper [here](https://github.com/penumbra-zone/web/pull/1279/files#diff-09f5d1ff494b1453dec33b98e4f5dad528c265e744fd7ce3341139e5ae164874R42-R74).

**Pull request merged**